nipype/utils/filemanip.py

```diff
@@ -702,8 +702,8 @@ def savepkl(filename, record, versioning=False):
         except FileNotFoundError as e:
             fmlogger.debug(str(e))
             sleep(2)
-    else:
-        raise e
+        else:
+            raise
```

Comment:
This was meant to be a `for`/`else`, not a `try`/`except`/`else`. What we need to do is instead:

```python
for ...:
    try:
        ...
        break
    except FileNotFoundError as e:
        err = e
        ...
else:
    raise err
```

Reply:
By the way, `sleep(2)` should be skipped in the last iteration.